File: Temerature Convertor using C++/temperatureConvertor.cpp
// Program to make temperature convertor

#include<iostream>

using namespace std;

float celcius, fahrenheit, kelvin;

// Function to convert celcius to fahrenheit
void celcius_fahrenheit()
{
    cout << "Enter the temperature in Celcius: ";
    cin >> celcius;

    fahrenheit = ((celcius * 9)/5) + 32;
    cout << "Temperature in Fahrenheit: " << fahrenheit << endl;
}

// Function to convert fahrenheit to celcius
void fahrenheit_celcius()
{
    cout << "Enter the temperature in Fahrenheit: ";
    cin >> fahrenheit;

    celcius = ((fahrenheit - 32) * 5) / 9;
    cout << "Temperature in Celcius: " << celcius << endl;
}

// Function to convert kelvin to celcius
void kelvin_celcius()
{
    cout << "Enter the temperature in Kelvin: ";
    cin >> kelvin;

    celcius = (kelvin - 273.15);
    cout << "Temperature in Celcius: " << celcius << endl;
}

// Function to convert celcius to kelvin
void celcius_kelvin()
{
    cout << "Enter the temperature in Celcius: ";
    cin >> celcius;

    kelvin = celcius + 273.15;
    cout << "Temperature in Kelvin: " << kelvin << endl;
}

// Function to convert kelvin to farenheit
void kelvin_fahrenheit()
{
    cout << "Enter the temperature in Kelvin: ";
    cin >> kelvin;

    fahrenheit = (((kelvin - 273.15) * 9) / 5) + 32;
    cout << "Temperature in Fahrenheit: " << fahrenheit << endl;
}

// Function to convert fahrenheit to kelvin
void fahrenheit_kelvin()
{
    cout << "Enter the temperature in Fahrenheit: ";
    cin >> fahrenheit;

    kelvin = (((fahrenheit - 32) * 5) / 9 ) + 273.15;
    cout << "Temperature in Kelvin: " << kelvin << endl;
}

int main()
{
    int choice = 1;
    while(choice != 7)
    {
        cout << "\n*** Temperature Convertor ***" << endl;
        cout << "1. Celcius to Fahrenheit" << endl;
        cout << "2. Fahrenheit to Celcius" << endl;
        cout << "3. Kelvin to Celcius" << endl; 
        cout << "4. Celcius to Kelvin" << endl;
        cout << "5. Kelvin to Fahrenheit" << endl;
        cout << "6. Fahrenheit to Kelvin" << endl; 
        cout << "7. EXIT" << endl;

        cout << "Enter Your Choice: ";
        cin >> choice;

        switch(choice)
        {
            case 1:
            {
                celcius_fahrenheit();
                break;
            }

            case 2:
            {
                fahrenheit_celcius();
                break;
            }

            case 3:
            {
                kelvin_celcius();
                break;
            }

            case 4:
            {
                celcius_kelvin();
                break;
            }

            case 5:
            {
                kelvin_fahrenheit();
                break;
            }

            case 6:
            {
                fahrenheit_kelvin();
                break;
            }

            default:
            {
                if(choice > 7)
                {
                    cout << "\nEnter Right Choice !! " << endl;
                }
            }
        }

    }

    return 0;    
}

File: Temperature Convertor Using Java/temperatureConvertor.java
import java.util.*;

public class temperatureConvertor {
    
    // function to convert celcius to fahrenheit
    public static void celciusFahrenheit() {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the temperature in celcius: ");
        float celcius = input.nextFloat();
            
        float fahrenheit = ((celcius * 9) / 5) + 32;
        System.out.println("Temperature in Fahrenheit: "+ fahrenheit);
        
        input.close();
    }

    // function to convert fahrenheit to celcius
    public static void fahrenheitCelcius() {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the temperature in fahrenheit: ");
        float fahrenheit = input.nextFloat();

        float celcius = ((fahrenheit - 32) * 5) / 9;
        System.out.println("Temperature in Celcius: "+ celcius);
        
        input.close();
    }
    

    // function to convert kelvin to celcius
    public static void kelvinCelcius() {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the temperature in kelvin: ");
        float kelvin = input.nextFloat();

        double celcius = (kelvin - 273.15);
        System.out.println("Temperature in Celcius: "+ celcius);
        
        input.close();
    }

    // function to convert celcius to kelvin
    public static void celciusKelvin() {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the temperature in Celcius: ");
        float celcius = input.nextFloat();

        double kelvin = (celcius + 273.15);
        System.out.println("Temperature in Kelvin: "+ kelvin);
        
        input.close();
    }

    // function to convert kelvin to fahrenheit
    public static void kelvinFahrenheit() {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the temperature in Kelvin: ");
        float kelvin = input.nextFloat();

        double fahrenheit = (((kelvin - 273.15) * 9) / 5) + 32;
        System.out.println("Temperature in Fahrenheit: "+ fahrenheit);
        
        input.close();
    }

    // function to convert fahrenheit to kelvin
    public static void fahrenheitKelvin() {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the temperature in Fahrenheit: ");
        float fahrenheit = input.nextFloat();

        double kelvin = (((fahrenheit - 32) * 5) / 9) + 273.15;
        System.out.println("Temperature in Kelvin: "+ kelvin);
        
        input.close();
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int choice = 1;
        while(choice != 7) {
            System.out.println("***Temperature Convertor***");
            System.out.println("1. Celcius to Fahrenheit");
            System.out.println("2. Fahrenheit to Celcius");
            System.out.println("3. Kelvin to Celcius");
            System.out.println("4. Celcius to Kelvin");
            System.out.println("5. Kelvin to Fahrenheit");
            System.out.println("6. Fahrenheit to Kelvin");
            System.out.println("7. Exit");
    
    
            System.out.print("Enter Your Choice: ");
            choice = input.nextInt();

            switch(choice) {
                case 1:
                    celciusFahrenheit();
                    break;
                case 2:
                    fahrenheitCelcius();
                    break;
                case 3:
                    kelvinCelcius();
                    break;
                case 4:
                    celciusKelvin();
                    break;
                case 5:
                    kelvinFahrenheit();
                    break;
                case 6:
                    fahrenheitKelvin();
                    break;
                case 7:
                    System.out.println("Thanks For Using");
                    break;
                default:
                    System.out.println("!! Enter Right Choice !!");
            }
        }
        input.close();
    }
}
